lib/ProductOpener/Display.pm

```diff
@@ -1532,10 +1504,69 @@ sub query_list_of_tags($$) {
 		}
 		else {
 			$log->debug("No results for aggregate MongoDB query key", { key => $key }) if $log->is_debug();
+		}
+	}
+
+	# If it is the first page and the number of results we got is inferior to the limit
+	# we do not need to count the results
+
+	my $number_of_results;
+
+	if (defined $results) {
+		$number_of_results = scalar @{$results};
+		$log->debug("MongoDB query results count", { number_of_results => $number_of_results }) if $log->is_debug();
+	}
+
+	if (($skip == 0) and (defined $number_of_results) and ($number_of_results < $limit)) {
+			$request_ref->{structured_response}{count} = $number_of_results;
+			$log->debug("Directly setting structured_response count", { number_of_results => $number_of_results }) if $log->is_debug();
+	}
+	else {
 
+		#get total count for aggregate (without limit) and put result in cache
+		my $key_count = $server_domain . "/" . freeze($aggregate_count_parameters);
+		$log->debug("MongoDB aggregate count query key", { key => $key_count }) if $log->is_debug();
+		$key_count = md5_hex($key_count);
+		my $results_count = get_cache_results($key_count,$request_ref);
+
+		if (not defined $results_count) {
+
+			my $count_results;
+
+			# do not used the smaller cached products_ tags collection if ?nocache=1
+			# or if the user is logged in and nocache is different from 0
+			if ( ((defined $request_ref->{nocache}) and ($request_ref->{nocache}))
+				or ((defined $User_id) and not ((defined $request_ref->{nocache}) and ($request_ref->{nocache} == 0)))
+				) {
+				eval {
+					$log->debug("Executing MongoDB aggregate count query on products_tags collection", { query => $aggregate_count_parameters }) if $log->is_debug();
+					$count_results = execute_query(sub {
+						return get_products_tags_collection()->aggregate( $aggregate_count_parameters, { allowDiskUse => 1 } );
```

Comment:
I think it is the opposite : if nocache=1 then products collection is used else it should be products_tags.

Reply:
Good catch, I'll fix it.

Reply:
Fixed.